[WIP] The Road to Compiling (Sponge branch)

hsyyid opened the pull request:
This PR will (eventually) make WorldGuard for Sponge compile.

Any input is welcome. :smile:

hsyyid committed:
Make Shizzles Compile.

Un-mess with POM.XML

More fixes.

Really, just use Gradle!

Stuff

Unmess with POM again

And again..

Forgot this bit

Conflicts are bad

Removed Non-Needed Classes

Anti-Formatting

More Stuffs

Formatting bad

Continue fixing shizzles

POM.XML

POM.XML #2

Change events

Stuff

More fixes

More fixes

More fies

Switch to correct Event.

clienthax commented on src/main/java/com/sk89q/worldguard/sponge/listener/BlacklistListener.java:
```diff
@@ -19,9 +19,21 @@
 package com.sk89q.worldguard.sponge.listener;
 
+<<<<<<< HEAD
+import static com.sk89q.worldguard.sponge.util.WorldEditTransforms.toVector;
```
Derp

clienthax commented on src/main/java/com/sk89q/worldguard/chest/SignChestProtection.java:
```diff
@@ -19,7 +19,10 @@
+<<<<<<< HEAD
+=======
 import java.util.Optional;
+>>>>>>> upstream/sponge
```
derp

clienthax commented on src/main/java/com/sk89q/worldguard/protection/flags/LazyLocation.java:
```diff
@@ -20,12 +20,17 @@
+<<<<<<< HEAD
+=======
 import java.util.Optional;
+>>>>>>> upstream/sponge
```
derp

clienthax commented on src/main/java/com/sk89q/worldguard/protection/flags/IntegerFlag.java:
```diff
@@ -36,7 +36,11 @@ public IntegerFlag(String name) {
+<<<<<<< HEAD
+    public Integer parseInput(WorldGuardPlugin plugin, CommandSource sender,String input) throws InvalidFlagFormat {
+=======
     public Integer parseInput(WorldGuardPlugin plugin, CommandSource sender, String input) throws InvalidFlagFormat {
```
derp

clienthax commented on src/main/java/com/sk89q/worldguard/protection/flags/EnumFlag.java:
```diff
@@ -19,12 +19,16 @@
 
+<<<<<<< HEAD
+=======
 import java.util.Optional;
```
derp

bloodmc commented:
@hsyyid squash some of these commits

hsyyid commented:
@bloodmc Okay, I'll squash em all

bloodmc commented on src/main/java/com/sk89q/worldguard/protection/managers/RegionManager.java:
```diff
@@ -54,396 +54,450 @@
-    public ApplicableRegionSet getApplicableRegions(Location loc) {
-        return getApplicableRegions(loc.getPosition());
-    }
+public final class RegionManager
```
Why did you mess with the formatting on this class?

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/BlacklistListener.java:
```diff
@@ -224,7 +235,7 @@ public void onUseItem(UseItemEvent event) {
 
     @Listener
-    public void onPlayerDropItem(DropItemStackEvent.Pre event) {
+    public void onPlayerDropItem(DropItemEvent.Dispense event) {
```
This should be DropItemEvent to handle Destruct too

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/debounce/BlockBlockPair.java:
```diff
@@ -19,14 +19,14 @@
     private final BlockMaterialPair block2;
 
-    public BlockBlockPair(Block block1, Block block2) {
+    public BlockBlockPair(BlockSnapshot block1, BlockSnapshot block2) {
```
This class isn't needed. Use 

```
Transaction<BlockSnapshot>
```

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/debounce/BlockMaterialPair.java:
```diff
@@ -19,20 +19,20 @@
+    private final BlockType material;
 
-    public BlockMaterialPair(Block block) {
+    public BlockMaterialPair(BlockSnapshot block) {
```
This class isn't needed, you just need a single BlockSnapshot.

bloodmc commented on pom.xml:
```diff
@@ -447,4 +447,4 @@
     </profile>
   </profiles>
-</project>
+</project>
```
Unneeded formatting change

bloodmc commented on src/main/java/com/sk89q/worldguard/protection/managers/index/HashMapIndex.java:
```diff
@@ -291,4 +277,25 @@ public HashMapIndex get() {
         }
     }
 
+	@Override
```
Fix formatting

hsyyid commented:
@bloodmc I have no clue why it's doing that to the POM.XML, must be something dumb about my IDE.

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/BlacklistListener.java:
```diff
@@ -254,70 +261,81 @@ public void onBlockDispense(BlockDispenseEvent event) {
-
-                if (inventory.getHolder().equals(player)) {
-                    event.setCurrentItem(null);
+    public void onInventoryClick(InteractInventoryEvent event) {
```
This should be ClickInventoryEvent

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/debounce/legacy/BlockEntityEventDebounce.java:
```diff
@@ -32,18 +32,18 @@ public BlockEntityEventDebounce(int debounceTime) {
-        private final Block block;
-        private final Material blockMaterial;
+        private final BlockSnapshot block;
+        private final BlockType blockType;
```
This should be removed as it is already part of BlockSnapshot.

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/RegionProtectionListener.java:
```diff
@@ -286,9 +297,8 @@ public boolean apply(Location target) {
     }
 
-    @EventHandler(ignoreCancelled = true)
+    @Listener
```
This listener should handle multiple entity spawns, not just one.

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/RegionProtectionListener.java:
```diff
@@ -333,13 +343,12 @@ public void onSpawnEntity(SpawnEntityEvent event) {
     }
 
-    @EventHandler(ignoreCancelled = true)
+    @Listener
```
DestroyEntityEvent does not exist, this should be DestructEntityEvent

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/RegionProtectionListener.java:
```diff
@@ -190,9 +203,8 @@ public boolean apply(Location target) {
     }
 
-    @EventHandler(ignoreCancelled = true)
+    @Listener
```
BreakBlockEvent ? This no longer exists and should be ChangeBlockEvent.Break

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/RegionProtectionListener.java:
```diff
@@ -138,21 +152,20 @@ private RegionAssociable createRegionAssociable(Cause cause) {
     }
 
-    @EventHandler(ignoreCancelled = true)
+    @Listener
```
PlaceBlockEvent no longer exists, should be ChangeBlockEvent.Place

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/RegionProtectionListener.java:
```diff
@@ -229,13 +241,12 @@ public boolean apply(Location target) {
     }
 
-    @EventHandler(ignoreCancelled = true)
+    @Listener
```
UseBlockEvent no longer exists. Should be InteractBlockEvent.Secondary

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/RegionProtectionListener.java:
```diff
@@ -369,13 +378,12 @@ public void onDestroyEntity(DestroyEntityEvent event) {
     }
 
-    @EventHandler(ignoreCancelled = true)
+    @Listener
```
UseEntityEvent no longer exists, should be InteractEntityEvent.Secondary

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/WorldGuardBlockListener.java:
```diff
@@ -83,13 +87,15 @@ public void onBlockBreak(BreakBlockEvent event) {
-        Location<World> block = event.getCause().getFirst(Location.class).orNull();
-        if (block == null) return;
+    public void onBlockFromTo(final ChangeBlockEvent.Place event) {
+        Location<World> block = event.getCause().first(Location.class).orElse(null);
```
Location will never be in the cause, this should be looping through each

```
Transaction<BlockSnapshot>
```

hsyyid commented:
@bloodmc All of your comments are now fixed :smile:

bloodmc commented:
@hsyyid definitely not fixed, you need to fix RegionProtectionListener. Many events in that file do not exist.

hsyyid commented:
@bloodmc Erm... all I said was I fixed your comments - not I fixed everything. Also you commented on the wrong lines, making it look like it's not fixed.

bloodmc commented on src/main/java/com/sk89q/worldguard/sponge/listener/WorldGuardPlayerListener.java:
```diff
@@ -185,74 +181,82 @@ public void onPlayerLogin(ClientConnectionEvent.Auth event) {
-                        str.append(it.next().getId());
-                        if (it.hasNext()) {
-                            str.append(", ");
+    public void handleBlockRightClick(InteractBlockEvent event) {
```
This should be InteractBlockEvent.Secondary
